python/src/equistore/operations/empty_like.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Union
+
+from equistore import TensorBlock, TensorMap
+
+from . import _dispatch
+from ._utils import _check_parameters_in_gradient_block
+
+
+def empty_like(
+    tensor: TensorMap,
+    parameters: Union[List[str], str] = None,
+    requires_grad: bool = False,
+) -> TensorMap:
+    """Create an empty_like :py:class:`TensorMap` with the same metadata of the input
+    :py:class:`TensorMap` (tensor), but all the values unitilized.
+
+    All the blocks.values are unitialized matricies and the metadata are given as input.
+
+    :param tensor: input tensor from which you get the metadata.
+    :param parameters: which gradient parameter to copy. Default ``None``
+                      means copy all of them if present.
+                      Empty list for no gradients information.
+    :param requires_grad: If autograd should record operations on the returned tensor.
+                          Only for torch, default False.
+    """
+
+    blocks = []
+    for _k, block in tensor:
+        blocks.append(
+            empty_like_block(
+                block=block, parameters=parameters, requires_grad=requires_grad
+            )
+        )
+    return TensorMap(tensor.keys, blocks)
+
+
+def empty_like_block(
+    block: TensorBlock,
+    parameters: Union[List[str], str] = None,
+    requires_grad: bool = False,
+) -> TensorBlock:
+    """Create a empty_like :py:class:`TensorBlock` with the same metadata of the input
+    :py:class:`TensorBlock`, but all the values equal to zero.
+
+    :param block: input :py:class:`TensorBlock` from which you can get the matadata.
+    :param parameters: which gradient parameter to copy. Default ``None``
+                      means copy all of them if present.
+                      Empty list for no gradients information.
+    :param requires_grad: If autograd should record operations on the returned tensor.
+                          Only for torch, default False.
+    """
+
+    values = _dispatch.empty_like(block.values, requires_grad=requires_grad)
+    result_block = TensorBlock(
+        values=values,
+        samples=block.samples,
+        components=block.components,
+        properties=block.properties,
+    )
+
+    if isinstance(parameters, str):
+        parameters = [parameters]
+
+    if parameters is None:
+        parameters = block.gradients_list()
+    else:
+        _check_parameters_in_gradient_block(
+            block=block, parameters=parameters, fname="empty_like_block"
+        )
+
+    for parameter in parameters:
+        gradient = block.gradient(parameter)
+        gradient_data = _dispatch.empty_like(gradient.data)
+
+        result_block.add_gradient(
+            parameter,
+            gradient_data,
+            gradient.samples,
+            gradient.components,
+        )
+
+    return result_block
```

Comment:
I like that we know have a couple of helper functions that make these functions quite clear.